src/core/cgroup-bpf-ctx.c

```diff
@@ -1,22 +1,83 @@
 /* SPDX-License-Identifier: LGPL-2.1+ */
 
+#include "bpf-program.h"
 #include "cgroup-bpf-ctx.h"
 #include "cgroup.h"
 #include "memory-util.h"
 #include "strv.h"
 #include "string-util.h"
 #include "path-util.h"
+#include "strv.h"
+
+int bpffs_program_from_string(const char *str, enum bpf_attach_type *ret_attach_type, char **ret_bpffs_path) {
+        _cleanup_strv_free_ char **parts = NULL;
+        _cleanup_free_ char *dup = NULL;
+        int attach_type;
+
+        assert(str);
+        assert(ret_bpffs_path);
+        assert(ret_attach_type);
+
+        parts = strv_split(str, ":");
+        if (!parts)
+                return log_oom();
+
+        if (strv_length(parts) < 2)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "Unexpected BPF program string format %s", str);
+
+        dup = strdup(parts[1]);
+        if (!dup)
+                return log_oom();
+
+        attach_type = bpf_cgroup_attach_type_from_string(parts[0]);
+        if (attach_type < 0)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Unknown BPF attach type=%s", parts[0]);
+
+        *ret_attach_type = (enum bpf_attach_type) attach_type;
+        *ret_bpffs_path = TAKE_PTR(dup);
+        return 0;
+}
+
+int bpffs_program_to_string(enum bpf_attach_type attach_type, const char *bpffs_path, char **ret_str) {
+        _cleanup_free_ char *p = NULL;
+        const char *s = NULL;
+
+        assert(bpffs_path);
+        assert(ret_str);
+
+        s = bpf_cgroup_attach_type_to_string(attach_type);
+
+        /* Likely should update bpf_cgroup_attach_type_table with new mapping. */
+        if (!s)
+                s = "unknown";
```

Comment:
hmm, maybe just format the type as numeric integer here if we have no name for it? (and parse it the same way above?)

Reply:
addressed

Reply:
[Addressed](https://github.com/systemd/systemd/pull/13496/files#diff-11d3c4581343e0c3d561b27621161777ac95427feb2dd615b3c2f59429e80aefR128-R130)